actors/builtin/reward/reward_actor.go

```diff
@@ -139,6 +145,8 @@ func (a Actor) UpdateNetworkKPI(rt vmr.Runtime, currRealizedPower *abi.StoragePo
 		}
 
 		st.updateToNextEpochWithReward(*currRealizedPower)
+		// only update smoothed estimates after updating rewart
+		st.updateSmoothedEstimates(rt.CurrEpoch()-prev, rt.TotalFilCircSupply())
```

Comment:
Blocking: This is not nice. I am guessing this request came from a side-channel from the cryptoecon team and I missed it. It _may_ be an ok way of getting what we want, but it's nasty. Lacking a written design/motivation and discussion I don't think it is.

Firstly, the node cannot compute the circulating supply without consulting the reward actors for the total mined. We have somehow got to this point with that still not being recorded in state. We need to track the total mined first.

Secondly, it's backwards and arbitrary for the reward actor to be maintaining a smoothed version of something it got from the runtime that is calculated, in part, from values the VM (should have) received from it. I think this is an oblique attempt to smooth the impact of initial pledge fluctations on the computed money supply. But we should just smooth those values as inputs.

If we really wanted to smooth the money supply calculation directly, from non-smooth inputs, we could just require the VM to do it for us and move all this computation off-chain.

For now, remove this and revert this use of smoothing; go back to consulting rt.TotalFilCircSupply() where it's needed. The rest of this PR is then clear to land. Smoothing the supply is strictly lower priority than smoothing power shocks (for which being inputs to the epoch reward, and hence money supply, is the reason we're smoothing them anyway).

@zixuanzh

Reply:
I agree that smoothing circulating supply is lower priority than smoothing BR. Let's land the high priority changes first and do this as a further optimization.